File: C++/Insertion-sort.cpp
//insertion sort O(n^2) time notation but at the best cases it can reach O^n
void insertion(vector<int>& list, int limit) {
    for (int i = 1; i<limit; i++) {
        int j;
        int value = list[i];
        for (j = i-1; j>=0 && list[j]>value; j--) {
            swap(list[j+1], list[j]);
        }
    }
}

File: Javascript/Insertion-Sort.js
var Insertion = function(array) { 
  for (var i = 1; array.length > i; i++) {
    var value = array[i];
    for (var j = i-1; j >= 0 && array[j] > value; j--) {
      //swap 
      var temp = array[j+1];
      array[j+1] = array[j];
      array[j] = temp;
    }
  }
}
